core/modules/contextual/js/contextual.js

```diff
@@ -23,23 +23,110 @@ Backdrop.behaviors.contextualLinks = {
           return false;
         }
       );
+
       // Attach hover behavior to trigger and ul.contextual-links.
-      $trigger.add($links).on('hover',
-        function () { $region.addClass('contextual-links-region-active'); },
-        function () { $region.removeClass('contextual-links-region-active'); }
-      );
+      $trigger.add($links).on('mouseenter', function () {
+        $region.addClass('contextual-links-region-active');
+      });
+      $trigger.add($links).on('mouseleave', function () {
+        $region.removeClass('contextual-links-region-active');
+      });
+
       // Hide the contextual links when user clicks a link or rolls out of the .contextual-links-region.
       $region.on('mouseleave click', Backdrop.contextualLinks.mouseleave);
-      $region.on('hover',
-        function() { $trigger.addClass('contextual-links-trigger-active'); },
-        function() { $trigger.removeClass('contextual-links-trigger-active'); }
-      );
+      $region.on('mouseenter', function() {
+        $trigger.addClass('contextual-links-trigger-active');
+      });
+      $region.on('mouseleave', function() {
+        $trigger.removeClass('contextual-links-trigger-active');
+      });
+
       // Prepend the trigger.
       $wrapper.prepend($trigger);
-
-      // Prevent child contextual links overlapping their parent ones.
-      $wrapper.siblings().find('.contextual-links-wrapper').eq(0).css('margin-top', '20px');
     });
+
+    /**
+     * Adjusts trigger positions in contextual links to avoid overlaps.
+     */
+    function adjustContextualLinks() {
+      // Get all wrappers anywhere on the page and some info about each.
+      var allWrappers = [];
+      $('.contextual-links-wrapper', context).each(function() {
+        allWrappers.push({
+          'wrapper': $(this),
+          'regionOffsetBottom': $(this).parent().offset().top + $(this).parent().height(),
+          'hShift': 0,
+          'vShift': 0
+        });
+      });
+
+      // Reset margins on all wrappers.
+      allWrappers.forEach(function(info) {
+        info.wrapper.css('margin', '0');
```

Comment:
Then use this everywhere below where `.wrapper` was referenced:

```suggestion
        info.$wrapper.css('margin', '0');
```

Reply:
Done.